aws_doc_sdk_examples_tools/metadata.py

```diff
@@ -267,20 +318,24 @@ def from_yaml(
 
 def parse_services(
     yaml: Any, errors: MetadataErrors, known_services: Dict[str, Service]
-) -> Dict[str, List[str]]:
+) -> Dict[str, Set[str]]:
     if yaml is None:
         return {}
-    services: Dict[str, List[str]] = {}
+    services: Dict[str, Set[str]] = {}
     for name in yaml:
         if name not in known_services:
             errors.append(metadata_errors.UnknownService(service=name))
         else:
-            service: Dict[str, None] | None = yaml.get(name)
+            service: Dict[str, None] | Set[str] | None = yaml.get(name)
             # While .get replaces missing with {}, `sqs: ` in yaml parses a literal `None`
             if service is None:
-                service = {}
-            # Make a copy of the dict
-            services[name] = [*service.keys()]
+                service = set()
+            if isinstance(service, dict):
+                service = set(service.keys())
+            if isinstance(service, set):
```

Comment:
When is a service a set?

Reply:
When it's cross-service. Usually it's a set of one.